### DISCUSSION TITLE
Any plans to move to newest Flutter SDK?

### DISCUSSION THREAD

**Discussion by smoyerx** [powersync-ja/sqlite_async.dart#9] [General]
Looks like an interesting approach vs sqflite_common_ffi and I'm looking forward to giving it a spin!  Any plans to move to the latest Flutter SDK series (3.x)?

**Comment by rkistner** [comment]
Do you mean Dart 3.x? This is library is already compatible with that. I'm adding explicit tests for that now in #10.

**Comment by smoyerx** [comment]
Right, I should have said Dart 3.x (my current project is on Flutter 3.10.6 / Dart 3.0.6 stables).  The pubspec.yaml for sqlite_async has SDK '>=2.19.1 <3.0.0', versus the more recent '>=3.0.0 <4.0.0'.  I believe this is why the sqlite_async pubspec.yaml also specifies an earlier version of sqlite3, specifically ^1.10.1, whereas my project's pubspec.yaml specifies sqlite3 ^2.0.0.

The net is that 'dart pub add sqlite_async' fails in my environment due to dependency mismatch.

I can set up a down rev environment to play with sqlite_async, no problem.  But I was curious if you plan to update the dependencies in a future release.

**Reply by rkistner** [reply]
According to [this](https://dart.dev/resources/dart-3-migration#dart-3-backwards-compatibility), it should already work with Dart 3.x, despite how the version is specified. But sqlite3 2.0.0 would cause an issue - I'll release a new version that is compatible soon.

**Comment by smoyerx** [comment]
OK, I was not aware of this Dart 3 migration support.  I'm new to the environment and started with 3.x.  So then it is purely the sqlite3 version causing the issue.  Thanks for your response.

**Comment by rkistner** [comment]
Version 0.5.0 is now released and has updated dependencies for both dart and sqlite3

**Comment by smoyerx** [comment]
I confirmed that I can now 'dart pub add sqlite_async' to my dev environment.  I look forward to trying it out!  Many thanks!  :-)
